### ISSUE TITLE
Check: CKV_AWS_45: Ensure no hard-coded secrets exist in Lambda environment false-positive

### ISSUE THREAD

**Issue by caevv** [bridgecrewio/checkov#6700]
**Describe the issue**
I have a lambda that is passing the secret name env variable, but not the secret value, and is failing check `CKV_AWS_45: "Ensure no hard-coded secrets exist in Lambda environment"`

**Examples**
CDK:
```ts
const githubSecret = new aws_secretsmanager.Secret(this, 'secret', {
  secretName: 'my-secret',
})

const lambda = new aws_lambda.GoFunction(this, "lambda", {
  runtime: aws_lambda.Runtime.PROVIDED_AL2023,
  timeout: Duration.minutes(1),
  entry: `../cmd`,
  architecture: aws_lambda.Architecture.ARM_64,
  environment: {
    APP_PRIVATE_KEY_SECRET_NAME: githubSecret.secretName,
  }
})
githubSecret.grantRead(lambda)
```
**Version:**
```
❯ checkov --version
3.2.120
```

**Comment by tsmithv11** [comment]
Hey @caevv thank you for reporting. I'm unfortunately not able to replicate this. Can you grab a log or screenshot of the full Checkov output for this finding so we can debug and fix it?

**Comment by kalifs** [comment]
When I use`"TAGS": git.commit.sha:55e7e7703f17c41f276caf8f1a1b744d674259f8` then the `Check: CKV_AWS_45: "Ensure no hard-coded secrets exist in Lambda environment"` is displayed.
When I only use SHA but not `git.commit.sha:` then the error goes away.
